### PR TITLE
fix: ensure CLI returns correct exit code (#626)

### PR DESCRIPTION
**Pull request by eysi09**
...also changed LoggerType from enum to union type

### CODE REVIEW

File: garden-service/src/cli/cli.ts [hunk 1]
```diff
@@ -97,22 +97,23 @@ export const MOCK_CONFIG: GardenConfig = {
...
   }),
-  env: new EnvironmentOption(),
-  loggerType: new ChoicesParameter({
-    choices: Object.keys(WRITER_CLASSES),
-    help: `TODO`,
+  "env": new EnvironmentOption(),
+  "logger-type": new ChoicesParameter({
+    choices: [...LOGGER_TYPES],
+    defaultValue: DEFAULT_CLI_LOGGER_TYPE,
+    help: "Set logger type.",
```
Comment (thsig) [inline garden-service/src/cli/cli.ts]: I think we should include descriptions of the available logger types. Maybe something like this? Not too happy about the formatting, but we haven't been using line breaks in these descriptions so far.

> Set logger type: fancy: updates log lines in-place when their status changes (e.g. when tasks complete), basic: appends a new log line when a log line's status changes, quiet: suppresses all log output.

**Comment by thsig** [comment]
Looks good! Had one comment/thought about the docstring for the `logger-type` option.

**Comment by eysi09** [comment]
Since we're changing the flag, it's technically a breaking change. However, in the previous version the flag wasn't documented at all, the help text was simply: `TODO`. So I guess we can let it slide.

File: garden-service/src/cli/cli.ts (hunk 1 above)
Comment (eysi09) [reply garden-service/src/cli/cli.ts]: There's also the `json` type, any thoughts on that one?

**Pull request merged**